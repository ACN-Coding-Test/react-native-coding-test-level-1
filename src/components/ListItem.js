import React from 'react';
import PropTypes from 'prop-types'
import { StyleSheet, View, Text } from 'react-native';
import theme from "../theme";

function ListItem(props) {
    const { title, subTitle, containerStyle, ...otherProps } = props;
    return (
        <View style={[styles.containerStyle, containerStyle]}>
            <Text style={styles.titleStyle}>{title}</Text>
            <Text style={styles.subTitleStyle}>{subTitle}</Text>
        </View>
    );
}

ListItem.defaultProps = {
    title: "",
    subTitle: ""
}

ListItem.propTypes = {
    title: PropTypes.string,
    subTitle: PropTypes.string
}


const styles = StyleSheet.create({
    containerStyle: {
        width: "100%",
        padding: theme.padding.default,
    },
    titleStyle: {
        fontSize: theme.fontSizes.body,
        color: theme.colors.primary
    },
    subTitleStyle: {
        fontSize: theme.fontSizes.subTitle
    }
});


export default ListItem;
