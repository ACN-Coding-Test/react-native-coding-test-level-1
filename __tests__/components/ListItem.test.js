import React from "react";
import renderer from "react-test-renderer";
import ListItem from "../../src/components/ListItem";

describe("ListItem:", () => {
  it("should render correctly", () => {
    const tree = renderer.create(<ListItem />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
