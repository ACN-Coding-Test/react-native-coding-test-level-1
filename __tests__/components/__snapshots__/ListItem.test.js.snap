// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem: should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "padding": 15,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#A100FF",
        "fontSize": 20,
      }
    }
  >
    
  </Text>
  <Text
    style={
      Object {
        "fontSize": undefined,
      }
    }
  >
    
  </Text>
</View>
`;
